tests/android/device.spec.ts

```diff
@@ -57,3 +58,41 @@ test('androidDevice.fill', async function({ androidDevice }) {
   await androidDevice.fill({ res: 'org.chromium.webview_shell:id/url_field' }, 'Hello');
   expect((await androidDevice.info({ res: 'org.chromium.webview_shell:id/url_field' })).text).toBe('Hello');
 });
+
+function wait(ms: number) {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve('');
+    }, ms);
+  });
+}
+
+test.only('androidDevice.options.autoInstallDriver', async function({ playwright }) {
+  const devices = await playwright._android.devices({ autoInstallDriver: false });
+
+  const androidDevice = devices[0];
+  await androidDevice.shell(`cmd package uninstall com.microsoft.playwright.androiddriver`);
+  await androidDevice.shell(`cmd package uninstall com.microsoft.playwright.androiddriver.test`);
+
+  await androidDevice.shell('am start -n com.android.chrome/com.google.android.apps.chrome.Main about:blank');
+
+  let fillEnd = false;
+  const fillPromise = androidDevice.fill({ res: 'com.android.chrome:id/url_bar' }, 'Hello').then(() => {
+    fillEnd = true;
+  });
+
+  await wait(2000);
```

Comment:
We only wait in tests to simulate race conditions, otherwise tests need to be predictable.